src/crt/sllshrs.src

```diff
@@ -0,0 +1,49 @@
+; Performs signed 48 bit ("short long long") bit shift right
+; 
+; Arguments:
+; ude = Most significant bytes of arg 1
+; uhl = Least significant bytes of arg 1
+; ubc = Amount to shift by (effectively just shift by c)
+
+
+; Returns:
+; ude:uhl = ude:uhl>>c
+
+	assume	adl=1
+
+	section	.text
+	public	__sllshrs
+__sllshrs:
+	push bc
+	ld b, c
+	push de
+	push hl
+	ld hl, 0
```

Comment:
You could make this a `ld hl,5 \ add hl,sp` and move it into the loop to save space, since that has the same cost as 5 increments.